### PR TITLE
Keep headers private

### LINKED ISSUE #13: *.h files from contrib/ directory shouldn't be copied into package
In `package()` method all header files are copied from source folder recursively, including that:

    ./zlib-1.2.8/contrib/blast/blast.h
    ./zlib-1.2.8/contrib/infback9/infback9.h
    ./zlib-1.2.8/contrib/infback9/inffix9.h
    ./zlib-1.2.8/contrib/infback9/inflate9.h
    ./zlib-1.2.8/contrib/infback9/inftree9.h
    ./zlib-1.2.8/contrib/iostream/zfstream.h
    ./zlib-1.2.8/contrib/iostream2/zstream.h
    ./zlib-1.2.8/contrib/iostream3/zfstream.h
    ./zlib-1.2.8/contrib/minizip/crypt.h
    ./zlib-1.2.8/contrib/minizip/ioapi.h
    ./zlib-1.2.8/contrib/minizip/iowin32.h
    ./zlib-1.2.8/contrib/minizip/mztools.h
    ./zlib-1.2.8/contrib/minizip/unzip.h
    ./zlib-1.2.8/contrib/minizip/zip.h
    ./zlib-1.2.8/contrib/puff/puff.h

But they don't needed, and most important, their presence will hide lack of (for example) minizip library until linking.

As `conans.ConanFile.copy()` method doesn't have option for non-recursive copy, maybe best solution will be to delete `contrib/` folder in `source()` method.

### PR DESCRIPTION
**Pull request by liberforce**
closes https://github.com/conan-community/community/issues/61

Fixes #13 as well as minor coding style and wrong order in calling CMake project() function.

### CODE REVIEW

**Comment by liberforce** [comment]
Appveyor builds failed and I see things like this in [the logs](https://ci.appveyor.com/project/lasote/conan-zlib-7h7rb/build/1.0.57/job/36g7fldc9oak1p0m)

```
Renaming
--     C:/Users/appveyor/.conan/data/zlib/1.2.11/conan/testing/build/8070b35f539f5318d470a276d7bb0f9f07ff0d40/zlib-1.2.11/zconf.h
-- to 'zconf.h.included' because this file is included with zlib
-- but CMake generates it automatically in the build directory.
```
Let's try removing zconf.h too.

**Comment by liberforce** [comment]
Build breaks on Linux is zconf is not provided, let's dig a bit more...

File: conanfile.py [hunk 1]
```diff
@@ -87,9 +87,9 @@ def package(self):
         self.copy("*.pc", dst="", keep_path=False)
 
         # Copy headers
-        for header in ["zlib.h", "_build/zconf.h"]:
-            self.copy(header, "include", "%s" % self.ZIP_FOLDER_NAME, keep_path=False)
-            self.copy(header, "include", "%s" % "_build", keep_path=False)
+        for header in ["*zlib.h", "*zconf.h", "*zutil.h"]:
```
Comment (liberforce) [inline conanfile.py]: Are you sure including `zutil.h` is useful ?

At the top of the file, it has this:

```
/* WARNING: this file should *not* be used by applications. It is
   part of the implementation of the compression library and is
   subject to change. Applications should only use zlib.h.
 */
```

It's not included in the Ubuntu package either:
https://packages.ubuntu.com/bionic/amd64/zlib1g-dev/filelist

**Comment by liberforce** [comment]
@SSE4 : Thanks for trying to fix that. It went out of my scope, and I got bored of how slow the appveyor tests where running.

File: conanfile.py (hunk 1 above)
Comment (SSE4) [reply conanfile.py]: I think I'll remove it then. I've left it because comment mentioned this.

**Pull request merged**